### ISSUE TITLE
sinfo doesn't show correct info on new login nodes

### ISSUE THREAD

**Issue by themisa** [clustervision/trinity#585]
Even after restarting slurm:

from /var/log/slurm/slurmctld.log

```
[2016-01-12T15:29:42.039] error: Unable to resolve "c003": Unknown host
[2016-01-12T15:29:42.039] error: fwd_tree_thread: can't find address for host c003, check slurm.conf
[2016-01-12T15:30:42.048] error: Unable to resolve "c002": Unknown host
[2016-01-12T15:30:42.048] error: fwd_tree_thread: can't find address for host c002, check slurm.conf
[2016-01-12T15:30:42.051] error: Unable to resolve "c003": Unknown host
[2016-01-12T15:30:42.051] error: fwd_tree_thread: can't find address for host c003, check slurm.conf
[2016-01-12T15:31:42.061] error: Unable to resolve "c002": Unknown host
[2016-01-12T15:31:42.061] error: fwd_tree_thread: can't find address for host c002, check slurm.conf
[2016-01-12T15:31:42.063] error: Unable to resolve "c003": Unknown host
[2016-01-12T15:31:42.063] error: fwd_tree_thread: can't find address for host c003, check slurm.conf
[2016-01-12T15:32:42.073] error: Unable to resolve "c002": Unknown host
[2016-01-12T15:32:42.073] error: fwd_tree_thread: can't find address for host c002, check slurm.conf
[2016-01-12T15:32:42.075] error: Unable to resolve "c003": Unknown host
[2016-01-12T15:32:42.075] error: fwd_tree_thread: can't find address for host c003, check slurm.conf
[2016-01-12T15:32:43.003] Terminate signal (SIGINT or SIGTERM) received
[2016-01-12T15:32:43.070] Saving all slurm state
[2016-01-12T15:32:44.195] error: Configured MailProg is invalid
[2016-01-12T15:32:44.196] slurmctld version 14.03.8 started on cluster (null)
[2016-01-12T15:32:44.205] error: Unable to resolve "c002": Unknown host
[2016-01-12T15:32:44.205] error: slurm_set_addr failure on c002
[2016-01-12T15:32:44.208] error: Unable to resolve "c003": Unknown host
[2016-01-12T15:32:44.208] error: slurm_set_addr failure on c003
[2016-01-12T15:32:44.208] Recovered state of 2 nodes
[2016-01-12T15:32:44.208] Recovered information about 0 jobs
[2016-01-12T15:32:44.208] Recovered state of 0 reservations
[2016-01-12T15:32:44.209] read_slurm_conf: backup_controller not specified.
[2016-01-12T15:32:44.209] Running as primary controller
[2016-01-12T15:33:44.221] SchedulerParameters=default_queue_depth=100,max_rpc_cnt=0,max_sched_time=4,partition_job_depth=0
[2016-01-12T15:36:24.137] Terminate signal (SIGINT or SIGTERM) received
[2016-01-12T15:36:24.254] Saving all slurm state
[2016-01-12T15:36:25.284] error: Configured MailProg is invalid
[2016-01-12T15:36:25.285] slurmctld version 14.03.8 started on cluster (null)
[2016-01-12T15:36:25.293] error: Unable to resolve "c002": Unknown host
[2016-01-12T15:36:25.293] error: slurm_set_addr failure on c002
[2016-01-12T15:36:25.296] error: Unable to resolve "c003": Unknown host
[2016-01-12T15:36:25.296] error: slurm_set_addr failure on c003
[2016-01-12T15:36:25.296] Recovered state of 2 nodes
[2016-01-12T15:36:25.296] Recovered information about 0 jobs
[2016-01-12T15:36:25.296] Recovered state of 0 reservations
[2016-01-12T15:36:25.297] read_slurm_conf: backup_controller not specified.
[2016-01-12T15:36:25.297] Running as primary controller
```

```
[root@login ~]# cat /etc/slurm/slurm-nodes.conf 
NodeName=c002 CPUS=32 State=UNKNOWN
NodeName=c003 CPUS=32 State=UNKNOWN
PartitionName=container Nodes=c002,c003 Default=Yes MaxTime=INFINITE State=UP
[root@login ~]# sinfo
sinfo: error: slurm_receive_msg: Zero Bytes were transmitted or received
slurm_load_partitions: Zero Bytes were transmitted or received
```

**Comment by themisa** [comment]
Note: the containers have restarted, and know where they belong to

```
[root@node002 ~]# docker ps
CONTAINER ID        IMAGE                     COMMAND                  CREATED             STATUS              PORTS               NAMES
fb5668f580e8        controller:5050/trinity   "/usr/bin/supervisord"   13 minutes ago      Up 13 minutes                           trinity
[root@node002 ~]# docker-enter trinity
[root@c002 /]# sinfo
PARTITION  AVAIL  TIMELIMIT  NODES  STATE NODELIST
container*    up   infinite      0    n/a 
[root@c002 /]# hostname
c002.vc-new
```

**Comment by hansthen** [comment]
Version?

**Comment by hansthen** [comment]
```
[root@login ~]# cat /etc/resolv.conf 
search cluster. vc-a. cluster
nameserver 10.141.255.254
```

**Comment by hansthen** [comment]
Fixed in 9b564c7f20

**Comment by themisa** [comment]
Fixed in
[root@login ~]# /trinity/report
**Version:** r7 b55e0e1
**Environment:** tmaster

For vc-new:
```
PARTITION  AVAIL  TIMELIMIT  NODES  STATE NODELIST
container*    up   infinite      2   idle c[001-002]
[root@login ~]# hostname
login
```
